File: 2021/09/01/main.go
package main

func arrayNesting(nums []int) int {
	r := 0
	for i := range nums {
		t := 1
		for nums[i] != i {
			nums[i], nums[nums[i]] = nums[nums[i]], nums[i]
			t++
		}
		if r < t {
			r = t
		}
	}
	return r
}

File: 2021/09/01/main.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        r = 0
        for i in range(len(nums)):
            if nums[i] != -1:
                a = nums[i]
                c = 0
                while nums[a] != -1:
                    t = a
                    a = nums[a]
                    c += 1
                    nums[t] = -1
                r = max(r, c)
        return r
